src/testController.ts

```diff
@@ -400,15 +400,20 @@ export class TestController {
     testItems.forEach((test) => {
       if (testItem) return;
 
-      if (test.children.size > 0) {
-        testItem = this.findTestByActiveLine(editor, test.children);
-      } else if (
+      if (
         test.uri?.toString() === editor.document.uri.toString() &&
         test.range?.start.line! <= line &&
         test.range?.end.line! >= line
       ) {
         testItem = test;
       }
+
+      if (test.children.size > 0) {
```

Comment:
Couldn't this be an `else if`? Am I misunderstanding the conditional on line 403?

Reply:
No, this is not possible, for example: 

```ruby
class Foo 
 
  def test_one # cursor here
    assert(true)
  end

end
```

The first condition will match (parent test range match the cursor) but you want to override the `testItem` value with child. Is it clearer?